src/corehost/cli/fxr/sdk_resolver.cpp

```diff
@@ -192,79 +249,374 @@ bool sdk_resolver_t::resolve_sdk_dotnet_path(
         }
     }
 
-    std::vector<pal::string_t> hive_dir;
-    get_framework_and_sdk_locations(dotnet_root, &hive_dir);
+    return {};
+}
 
-    pal::string_t cli_version;
-    pal::string_t sdk_path;
-    pal::string_t global_cli_version;
+bool sdk_resolver::parse_global_file(pal::string_t global_file_path)
+{
+    if (global_file_path.empty())
+    {
+        // Missing global.json is treated as success (use default resolver)
+        return true;
+    }
+
+    trace::verbose(_X("--- Resolving SDK information from global.json [%s]"), global_file_path.c_str());
 
-    if (!global.empty())
+    pal::ifstream_t file{ global_file_path };
+    if (!file.good())
     {
-        global_cli_version = resolve_cli_version(global);
+        trace::error(_X("[%s] could not be opened"), global_file_path.c_str());
+        return false;
     }
 
-    for (pal::string_t dir : hive_dir)
+    if (skip_utf8_bom(&file))
     {
-        trace::verbose(_X("Searching SDK directory in [%s]"), dir.c_str());
-        pal::string_t current_sdk_path = dir;
-        append_path(&current_sdk_path, _X("sdk"));
+        trace::verbose(_X("UTF-8 BOM skipped while reading [%s]"), global_file_path.c_str());
+    }
+
+    json_value doc;
+    try
+    {
+        doc = json_value::parse(file);
+    }
+    catch (const exception& ex)
+    {
+        pal::string_t msg;
+        (void)pal::utf8_palstring(ex.what(), &msg);
+        trace::error(_X("A JSON parsing exception occurred in [%s]: %s"), global_file_path.c_str(), msg.c_str());
+        return false;
+    }
+
+    if (!doc.is_object())
+    {
+        trace::error(_X("Expected a JSON object in [%s]"), global_file_path.c_str());
+        return false;
+    }
+
+    const auto& doc_obj = doc.as_object();
+
+    const auto sdk = doc_obj.find(_X("sdk"));
+    if (sdk == doc_obj.end() || sdk->second.is_null())
+    {
+        // Missing SDK is treated as success (use default resolver)
+        trace::verbose(_X("Value 'sdk' is missing or null in [%s]"), global_file_path.c_str());
+        return true;
+    }
 
-        if (global_cli_version.empty())
+    if (!sdk->second.is_object())
+    {
+        trace::error(_X("Expected a JSON object for the 'sdk' value in [%s]"), global_file_path.c_str());
```

Comment:
I'm fine with using warnings as we do continue to just fallback as if the global.json didn't exist.  I'll get that fixed.

Reply:
Given that we won't be able to plumb warnings to VS without [https://github.com/dotnet/core-setup/issues/6972](https://github.com/dotnet/core-setup/issues/6972), I think it would be good to report this as an error. 

---
In reply to: [298392322](https://github.com/dotnet/core-setup/pull/6953#discussion_r298392322) [](ancestors = 298392322)

Reply:
I missed this feedback from a few days ago.  Given @vitek-karas preferred a warning in this context, but we can't currently report them as warnings from the VS resolver, should we change back to errors until #6972 is implemented?

Reply:
Personally I'm not a big fan of writing to stderr without actually failing... but I get that in this case it might be helpful. The main reason I worry about this is things like msbuild - sometimes when msbuild runs a child process which writes to stderr it will consider the task a failure and fail the build (I think it depends on how the logging is setup for the child process).

One other consideration is if this is something we can change in the future. The code lives in `hostfxr` which is shared by all runtimes on the machine, so as such it is not exactly versioned. So even in .NET 5 we can't introduce breaking changes to it. In theory switching from stderr to stdout/warnings could be considered a breaking change.

I still think it would be better to keep these as warnings and tackle the issue of reporting warnings in VS.

Reply:
@nguerrera any strong opposition of merging this as-is and instead focusing on fixing #6972 to get the messages back?

Reply:
@vitek-karas this PR, as currently implemented, does include one error -> warning change, which is when global.json can't be parsed.  We used to error and ignore the file.  Now we warn about the parse failure and ignore the file with an additional warning saying the latest SDK will be used so users understand the file is being ignored.  Should we consider that breaking?

Reply:
I think that one is much less worrisome, but technically it is. The reason why this is much easier to accept is that if something watches stderr and marks everything there as error, it will simply miss this one, but since the process still runs - probably not a big deal.

That said, thanks for pointing it out, I missed it. I think we should probably revert that one change back. It's just very late in the release cycle, unless we have a good reason to break, we should not. Even if I don't like it... 😄

Reply:
Alright, I will restore any errors we were previously emitting when handling global.json.  Any new diagnostics that aren't failures will remain warnings.

Reply:
> @nguerrera any strong opposition of merging this as-is and instead focusing on fixing #6972 to get the messages back?

No opposition at all :)